### PR TITLE
restyle + refactor auth modals

### LINKED ISSUE #2022: Usage of white and transparent color in card assets looks a bit off
See https://github.com/votingworks/vxsuite/pull/1993#issuecomment-1162739340

### PR DESCRIPTION
**Pull request by adghayes**
## Overview
Closes #2022. Implements @beausmith's UI principle that purely action oriented screens (setup card reader, keypad, invalid card, remove card) are full screen modals and should all a) have a white background and b) have no election footer. 

Also does some work toward #893, abstracting the auth screens into shared UI components. When changing the screens, instead of changing it in both `bsd` and `election-manager`, I refactored to get the components from `libs/ui`

## Demo Video or Screenshot

Before:
[auth-ui-before.webm](https://user-images.githubusercontent.com/37960853/177234601-034e6710-89c2-408e-9e3a-fb53a1c2accc.webm)

After:
[auth-ui-tweak.webm](https://user-images.githubusercontent.com/37960853/177234616-04e7afb1-5366-4894-8b38-c107b6b832fb.webm)

## Testing Plan 
- [x] Updated tests in `frontends` for new copy
- [x] Moved `UnlockMachineScreen` tests to `libs/ui`
- [x] Added sanity check tests for new `libs/ui` components (not snapshots)

## Checklist
- [ ] ~~I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.~~
- [ ] I have added JSDoc comments to any newly introduced exports

- [x] I have added a screenshot and/or video to this PR to demo the change
- [x] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

**Comment by adghayes** [comment]
I should have split the refactor and the restyle into different commits but... alas. Luckily it isn't too much to review.

File: libs/ui/src/remove_card_screen.tsx
Comment (arsalansufi) [inline libs/ui/src/remove_card_screen.tsx:22]: Since we already have text with clear instructions, this image can probably be considered a decorative one from an accessibility standpoint, in which case we should make the `alt` tag an empty string, `alt=""`. Much more on this here! https://www.w3.org/WAI/tutorials/images/decorative

File: libs/ui/src/remove_card_screen.tsx
Comment (arsalansufi) [inline libs/ui/src/remove_card_screen.tsx:24]: Since this `p` just contains an image and no text, let's remove the `p` wrapping (and add styles as needed to `RemoveCardImage`)

File: libs/ui/src/invalid_card_screen.tsx
Comment (arsalansufi) [inline libs/ui/src/invalid_card_screen.tsx:11]: Any reason we apply `maxWidth={false}` to this screen but not the `RemoveCardScreen`?

File: libs/ui/src/unlock_machine_screen.tsx
Comment (arsalansufi) [inline libs/ui/src/unlock_machine_screen.tsx:34]: I believe we can also expect to receive an `InsertedSmartcardAuth.CheckingPasscode` auth object here. For example, on BMDs, admins will still use inserted smartcard auth instead of dipped smartcard auth. Likewise, poll worker cards will soon be able to have PINs (for VVSG2 cert).

(cc @jonahkagan)

File: libs/ui/src/invalid_card_screen.tsx
Comment (adghayes) [reply libs/ui/src/invalid_card_screen.tsx:11]: It looks like the `InvalidCardScreen` text will wrap if `maxWidth` is true. But the `RemoveCardScreen` text is too short to wrap even if `maxWidth` is true.

**Comment by adghayes** [comment]
Some funky test failures...

`precinct-scanner` test coverage not passing because the removal of the `unlock_admin_screen.tsx` page brought the average function coverage down. This seems like a case where our test coverage thresholds are not achieving anything. Do I just go find somewhere else unrelated to add a test to bring the coverage up?

`integration-testing-election-manager` is failing because of an assertion timeout. I suspect this would succeed if it ran again. Those tests are always so slow, I think we should look into making them faster.

File: libs/ui/src/invalid_card_screen.tsx
Comment (arsalansufi) [reply libs/ui/src/invalid_card_screen.tsx:11]: Got it!

**Comment by arsalansufi** [comment]
> precinct-scanner test coverage not passing because the removal of the unlock_admin_screen.tsx page brought the average function coverage down. This seems like a case where our test coverage thresholds are not achieving anything. Do I just go find somewhere else unrelated to add a test to bring the coverage up?

If the average test coverage percentage just dropped because of the removal of a fully covered file, I think it's okay to decrease the coverage threshold! Since we haven't actually regressed or increased the work to get to 100% test coverage

File: frontends/bsd/src/app.test.tsx
Comment (jonahkagan) [inline frontends/bsd/src/app.test.tsx:447]: One way I've seen inexact matching done in the testing library docs is to use a regex: `await screen.findByText(/Remove card/)`

I would personally lean towards just checking for the actual full text that's expected on the screen, but that's just me!

**Review by jonahkagan — COMMENTED** [review]
Love this refactor!

File: frontends/bsd/src/app.test.tsx
Comment (adghayes) [reply frontends/bsd/src/app.test.tsx]: I think of the full text as copy that could change (and did change between the original screen and this PR) and I wanted to avoid testing copy. I believe this copy is final, however, so checking the actual full text makes sense to me to avoid unintended consequences of inexact matching, so I switched it.

**Comment by adghayes** [comment]
@arsalansufi @jonahkagan this should be good to merge pending your approval.

**Review by arsalansufi — APPROVED** [review]
LGTM!

**Pull request merged**
